### PR TITLE
[GPU] Bugfix for accruacy of onednn reorder fusing

### PR DESCRIPTION
**Pull request by byungilm**
Signed-off-by: Min, Byungil <byungil.min@intel.com>

### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 1]
```diff
@@ -1107,6 +1113,17 @@ bool layout_optimizer::are_data_types_suitable_for_onednn(program_node& node) {
     return false;
 }
 
+bool layout_optimizer::are_layouts_suitable_for_onednn(program_node& node) {
+    auto in_layout = node.get_dependencies().front()->get_output_layout();
+    auto out_layout = node.get_output_layout();
+    if (node.get_preferred_impl_type() == impl_types::onednn && node.is_type<convolution>() &&
+        (in_layout.data_padding != out_layout.data_padding || out_layout.data_padding)) {
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:1120]: What about being more direct?
`in_layout.data_padding || out_layout.data_padding`?

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 2]
```diff
@@ -1107,6 +1113,17 @@ bool layout_optimizer::are_data_types_suitable_for_onednn(program_node& node) {
     return false;
 }
 
+bool layout_optimizer::are_layouts_suitable_for_onednn(program_node& node) {
+    auto in_layout = node.get_dependencies().front()->get_output_layout();
+    auto out_layout = node.get_output_layout();
+    if (node.get_preferred_impl_type() == impl_types::onednn && node.is_type<convolution>() &&
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:1119]: It is not only for convolution, but the entire onednn primitives.

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 3]
```diff
@@ -1107,6 +1113,17 @@ bool layout_optimizer::are_data_types_suitable_for_onednn(program_node& node) {
     return false;
 }
 
+bool layout_optimizer::are_layouts_suitable_for_onednn(program_node& node) {
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:1116]: What about `is_data_padded`, instead? Then `true` and `false` need to be swapped.

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 4]
```diff
@@ -308,7 +308,13 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
         // Remove Reorder for Convolution: b_fs_yx_fsv32 (i8/u8) -> b_fs_yx_fsv16 (fp32/fp16)
         if (next.is_type<convolution>() && fmt_prev == format::b_fs_yx_fsv32 && fmt_next == format::b_fs_yx_fsv16 &&
             !data_type_traits::is_floating_point(prev_dt) && data_type_traits::is_floating_point(next_dt)) {
-            return true;
+            auto& node = prev.get_users().front();
+            // Avoid to fuse padding reorder to previous onednn convolution
+            if (prev.is_type<convolution>() && prev.as<convolution>().get_preferred_impl_type() == impl_types::onednn &&
+                (node->get_output_layout().data_padding != prev.get_output_layout().data_padding))
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:314]: What about using `are_data_types_suitable_for_onednn` function here?

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp (hunk 1 above)
Comment (byungilm) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:1120]: looks good. fixed.

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp (hunk 2 above)
Comment (byungilm) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:1119]: Is it valid for reorder as well?

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp (hunk 4 above)
Comment (isanghao) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:314]: To be followed up: it is not only for conv, but also for other onednn primitives as well.

**Review by isanghao — APPROVED** [review]
Please see comment

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp (hunk 4 above)
Comment (byungilm) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:314]: This will be applied to next PR with general reorder removal together.

**Pull request merged**